### PR TITLE
many: remove snappy.Part totally and replace with snap.Info

### PR DESCRIPTION
**Pull request by mvo5**
[based on https://github.com/ubuntu-core/snappy/pull/655]

https://github.com/mvo5/snappy/compare/refactor/no-parts4...mvo5:refactor/no-parts5

This branch removes the use of snappy.Part entirely. There is a bit of the concept behind it still lurking that needs killing still.

### CODE REVIEW

**Comment by mvo5** [comment]
retest this please

**Comment by chipaca** [comment]
How is `BaseSnap` different from `SnapInfo`?

**Comment by mvo5** [comment]
@chipaca heh, that is an excellent point, indeed, we should try to make this leap already here.

**Comment by mvo5** [comment]
retest this please

**Comment by mvo5** [comment]
retest this please

**Comment by zyga** [comment]
retest this please

**Comment by mvo5** [comment]
retest this please

File: snap/info.go [hunk 1]
```diff
@@ -21,7 +21,10 @@ package snap
 
 // Info provides information about packages
```
Comment (niemeyer) [inline snap/info.go]: s/packages/snaps./

File: snap/info.go [hunk 2]
```diff
@@ -21,7 +21,10 @@ package snap
...
 type Info struct {
-	Name    string
-	Version string
-	Type    Type
+	Name        string
+	Developer   string
+	Version     string
+	Type        Type
+	Channel     string
```
Comment (niemeyer) [inline snap/info.go]: `Summary string`

File: snappy/snap_file.go [hunk 3]
```diff
@@ -83,6 +83,16 @@ func (s *SnapFile) Type() snap.Type {
 	return "app"
 }
 
+// Info returns the snap.Info data.
+func (s *SnapFile) Info() *snap.Info {
+	if info, err := s.deb.Info(); err == nil {
+		// developer is something that no
```
Comment (niemeyer) [inline snappy/snap_file.go]: Cropped comment.

**Comment by niemeyer** [comment]
LGTM

**Pull request merged**